qiskit/providers/ibmq/runtime/runtime_program.py

```diff
@@ -255,3 +258,77 @@ class ProgramResult(NamedTuple):
     name: str
     description: str
     type: str
+
+class ParameterNamespace(SimpleNamespace):
+    """ An abstraction for SimpleNamespace that offers param validation.
+
+    Note: Unused/Unnecessary params does not cause error (still validates)
+
+    Args:
+        params (List[ProgramParameter]): The program's input parameters
+
+    Raises
+        IBMQInputValueException on validation fail
+    """
+
+    def __init__(self, params: List[ProgramParameter]):
+        super().__init__()
+        # Allow access to the raw program parameters list
+        self.params_list = params
+        # For localized logic, create store of parameters in dictionary
+        self.__program_params: dict = {}
+
+        for param in params:
+            # (1) Add parameters to a dict by name
+            setattr(self, param.name, None)
+            # (2) Store the program params for validation
+            self.__program_params[param.name] = param
+
+    def validate(self) -> None:
+        """Validates the user's usage of the program's inputs
+
+        Raises:
+            IBMQInputValueError if validation fails
+        """
+
+        # Iterate through the user's stored inputs
+        for param_name, program_param in self.__program_params.items():
+            # Set invariants: User-specified parameter value (value) and whether it's required (req)
+            value = None
+            req = program_param.required
+            # Check there exists a program parameter of that name.
+            try:
+                # Get the value currently stored for this parameter (user-adjusted)
+                value = self.__dict__[param_name]
+            except AttributeError:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+            # Check for empty value
+            if not value or value is None:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+
+            value_type = type(value).__name__
+            # Check program param of that name is of same type
+            if value is not None and value_type != program_param.type:
+                raise IBMQInputValueError('Parameter (%s) is incorrect type! (%s != %s)' % \
+                                          (param_name, value_type, program_param.type))
+
+    def __str__(self) -> str:
+        """Creates string representation of object"""
+
+        # ProgramParameter object (str)
+        def param_str(param: ProgramParameter) -> str:
+            return "------------\n<> Name        : %s\n<> Type        : %s\
+        \n<> Description : %s\n<> Required?   : %s\n" % \
+                    (param.name, param.type, param.description, param.required)
+
+        # List of ProgramParameter objects (str)
+        program_params_str = ''.join([param_str(param) for
+                                      param in self.__program_params.values()])
```

Comment:
I don't think it's quite necessary to have 5 lines per parameter. How about a tabular view, something like
```
Name   |  Value |  Required  | Type   | Description  
---------------------------------------------------
foo    |  3     |  True      | int    | Some number  
bar    |  None  |  True      | string | Some string  
```

I put name/value/required first because those are things users likely care about the most

Reply:
Fixed!